### PR TITLE
Update base image to be Debian 11 Bullseye

### PR DESCRIPTION
**Pull request by shreyaansjain06**
Fixes issue #11

### CODE REVIEW

**Comment by shreyaansjain06** [comment]
So I need to update debian such that all tests should pass right?
Have I understood it properly if not please tell what I am missing?

**Comment by jeffw16** [comment]
Yes, if you take a look at the detailed error message of this last CI build, it runs into an error on the line that does an apt install. This needs to be diagnosed and fixed.

The image needs to:


1. build properly
2. run without bugs

**Comment by shreyaansjain06** [comment]
I tried running the tests locally and they are failing without even modifying the code i.e. with detain:10
Screenshot 2022-03-04 at 12 43 11 PM

Should I try to resolve the errors there so that test works fine and then work on the updating Debian base image or I am doing any mistake in running the tests locally?

**Comment by shreyaansjain06** [comment]
Please ignore previous comment I found solution of that error

**Comment by shreyaansjain06** [comment]
I am not able to get why this test is failing during apt-install Can you help me by pointing to the specific thing due to which this error is occurring so that I can fix it

**Comment by jeffw16** [comment]
I am guessing your computer has ran out of memory. Two things point me to this hypothesis:

1. The exit code is 137. https://www.google.com/search?q=exit+code+137&oq=exit+code+137&aqs=chrome.0.0i512l10.4103j0j1&sourceid=chrome&ie=UTF-8
2. It says it was "killed".

**Comment by shreyaansjain06** [comment]
Thanks for your reply 
I have resolved Storage error by increasing storage 
but I am getting this error of Failure Login 
Screenshot 2022-03-05 at 4 13 41 PM

**Comment by jeffw16** [comment]
Hi, please use Google for these sorts of questions. https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

**Comment by shreyaansjain06** [comment]
Sorry for that but I googled and I did not got the solution 
I will not repeat this mistake in future

**Comment by shreyaansjain06** [comment]
It is not able to find the git version in tests when I changed debian:10 to debian:bullseye
```
#7 9.006 Unable to find a version "1:2.20.1-2+deb10u3" for the package "git"
```
I think for debian 11 bullseye the version of git needs to be changed to ```1:2.30.2-1``` [here](https://github.com/WikiWorks/Canasta/blob/298c97312f694903e77e09e08d6b5418016fc8a8/Dockerfile#L23)

Am I going in right direction or there may be some other cause of failure of the test?

**Comment by jeffw16** [comment]
@shreyaansjain06 Yeah the version will definitely need to be changed. Notice how the current version used has `+deb10u3` appended afterwards. Bullseye won't recognize this because its packages would need to contain `deb11` in the version.

**Comment by shreyaansjain06** [comment]
There is  `python3-pygments` in debian bullseye instead of `python-pygment` should I replace it?

**Comment by shreyaansjain06** [comment]
Thanks for your reply Jefrey

**Comment by shreyaansjain06** [comment]
I googled about this and found that this error is due to permissions of github token 
Either write permission can be given by UI or by Yaml file I have updated the file Should I push if you agree?

Below are the links supporting my point 
[link1](https://github.com/docker/build-push-action/issues/463#issuecomment-924995909)
[link2](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)

**Comment by jeffw16** [comment]
That's fine, one of the contributors to Canasta who has repo write access will have the ability to run the tests.

**Review by jeffw16 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -20,26 +20,26 @@ RUN set x; \
...
+	mc=3:4.8.26-1.1 \
+	ffmpeg=7:4.3.3-0+deb11u1 \
+	curl=7.74.0-1.3+deb11u1 \
+	unzip=6.0-26 \
+	gnupg=2.2.27-2 \
+	default-mysql-client=1.0.7 \
+	rsync=3.2.3-4+deb11u1 \
     lynx \
     && wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg \
     && echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/php.list \
```
[the remark this answers is not in the document]
Comment (shreyaansjain06) [reply Dockerfile:44]: Thanks for the review 
Removed the Sury's apt repository

[the remark this answers is not in the document]
Comment (jeffw16) [reply Dockerfile:44]: I don't believe you did. Maybe you forgot to push your changes to GitHub?

**Review by jeffw16 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -21,7 +21,7 @@ RUN set x; \
     && aptitude -y upgrade \
     && aptitude install -y \
     git=1:2.30.2-1 \
-    apache2=2.4.51-1~deb11u1 \
+    apache2=2.4.53-1~deb11u \
```
Comment (amalpaul54111) [inline Dockerfile:24]: I think here there is a typo, `2.4.53-1~deb11u1` is the correct version.

Comment (shreyaansjain06) [reply Dockerfile]: Yes you are right that was a typo

**Comment by shreyaansjain06** [comment]
Some packages version were updated, which was causing problem so I fixed them

**Review by jeffw16 — APPROVED** [review]
LGTM

**Pull request merged**